lib/diagnostic.rb

```diff
@@ -13,28 +13,29 @@ def longest(enum)
   # enumerable supplied (i.e. apply the `+` operator)
   # Assume each entry implements a `+` method
   def sum(enum)
-    # Your code goes here
+    enum.reduce(:+)
   end
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the numbers enumerable are odd
   def some_odd(numbers)
-    # Your code goes here
+    numbers.odd?
   end
 
   # Question 4
   # Write ruby code that returns true if every entry in the
   # numbers enumerable are even
   def every_even(numbers)
-    # Your code goes here
+    array = enum.each_with_object([]) { numbers.even? ? true : false }
+    array.include?(false) ? false : true
```

Comment:
A better approach would have been to use `.all?`:
`numbers.all? { |num| num.even? }`
(alternatively, `numbers.all?(&:even?)`)